pkg/kubernetes/reconcile.go

```diff
@@ -49,9 +87,33 @@ func Reconcile(raw map[string]interface{}, spec v1alpha1.Spec, targets []*regexp
 
 	// Stable output order
 	sort.SliceStable(out, func(i int, j int) bool {
```

Comment:
I'd like to have some tests for this at some point:
- Namespaces and CRD's before their instances
- Kinds not in list alphabetically
- equal Kinds by name

Not necessarily as part of this PR, choose at your liking

Reply:
I had a quick look at the current tests and it seems to be more appropriate to be implemented after https://github.com/grafana/tanka/issues/220 is fixed. I've added https://github.com/grafana/tanka/issues/248 to not forget about this.

Reply:
Actually it's simple enough for a basic unit test, but I'm happy with doing it separately